### ISSUE TITLE
If I consider Ca2+ feedback on rhodopsin kinasein in HSDM. Is there any simple method in IQM tool? Because in every time point IQMsimulate returns a Ca2+ value, IQMstochsim2 returns a set of value about RK, RECT, RecR_Ca,RecR_Ca2+_RK, which will cost too large calculation and time.

### ISSUE THREAD

**Issue by larowrow** [CBeelen/Phototransduction#2]

**Comment by CBeelen** [comment]
No, unfortunately there is no easy way to do this in HSDM, because of the separation of the stochastic frontend and the deterministic backend simulation. As you correctly point out, a number of variables would have to be handed back and forth between the two simulations, which is impractical.
In principle it would be possible to unify the model again and simulate the whole model stochastically, but this would be computationally very expensive because of the large number of interactions for Ca2+.
What you could also do is make a static approximation for the reactions involving the Recoverin, by making the reaction rates independent of Calcium. There is some more detail on this in my dissertation, which you can find here:
[PDF version](http://oops.uni-oldenburg.de/4738/1/beemod20.pdf)
Page 53 has some details on the static approximation. The reaction rates will only depend on the Rhodopsin kinase and not the calcium concentration. If you're however explicitly interested in the Calcium dependency, I don't know an easy way to implement that.
I'm leaving this one open as well for now, let me know if there are further questions.
